rules.go

```diff
@@ -521,159 +524,120 @@ It expects the config in a json formatted string of following format:
 */
 func SetRules(s Netlink, content []byte) error {
 	var (
-		rules      interface{}
-		err        error
-		strictPath bool
+		rules auditRules
+		err   error
 	)
 	err = json.Unmarshal(content, &rules)
 	if err != nil {
-		return errors.Wrap(err, "SetRules failed")
+		return err
 	}
-
-	m := rules.(map[string]interface{})
-
-	if err != nil {
-		return errors.Wrap(err, "SetRules failed")
-	}
-	if strict, ok := m["strict_path_check"]; ok && strict.(bool) {
-		strictPath = true
-	}
-	//TODO: syscallMap should be loaded according to runtime arch
+	// TODO: syscallMap should be loaded according to runtime arch
 	syscallMap := headers.SysMapX64
-	for k, v := range m {
-		auditSyscallAdded = false
-		switch k {
-		case "file_rules":
-			vi := v.([]interface{})
-			for ruleNo := range vi {
-				rule := vi[ruleNo].(map[string]interface{})
-				path, ok := rule["path"]
-				if path == "" || !ok {
-					return errors.Wrap(err, "SetRules failed: watch option needs a path")
-				}
-				var ruleData auditRuleData
-				ruleData.Buf = make([]byte, 0)
-				add := AUDIT_FILTER_EXIT
-				action := AUDIT_ALWAYS
-				auditSyscallAdded = true
-
-				err = auditSetupAndAddWatchDir(&ruleData, path.(string), strictPath)
-				if err != nil {
-					return errors.Wrap(err, "SetRules failed")
-				}
-				perms, ok := rule["permission"]
-				if ok {
-					err = auditSetupAndUpdatePerms(&ruleData, perms.(string))
-					if err != nil {
-						return errors.Wrap(err, "SetRules failed")
-					}
-				}
-
-				key, ok := rule["key"]
-				if ok {
-					err = auditRuleFieldPairData(&ruleData, key, AUDIT_EQUAL, "key", AUDIT_FILTER_UNSET) // &AUDIT_BIT_MASK
-					if err != nil {
-						return errors.Wrap(err, "SetRules failed")
-					}
-				}
-
-				err = auditAddRuleData(s, &ruleData, add, action)
-				if err != nil {
-					return errors.Wrap(err, "SetRules failed")
-				}
+	for _, x := range rules.FileRules {
+		var ruleData auditRuleData
+		ruleData.Buf = make([]byte, 0)
 
+		err = auditSetupAndAddWatchDir(&ruleData, x.Path, rules.StrictPathCheck)
+		if err != nil {
+			return err
+		}
+		err = auditSetupAndUpdatePerms(&ruleData, x.Permission)
+		if err != nil {
+			return err
+		}
+		fpd := fieldPairData{
+			fieldval:     x.Key,
+			opval:        AUDIT_EQUAL,
+			fieldname:    "key",
+			flags:        AUDIT_FILTER_UNSET,
+			syscallAdded: true,
+		}
+		err = auditRuleFieldPairData(&ruleData, &fpd)
```

Comment:
Ahhh yes, I see this value is optional. So I am assuming what we would want to do here, is only apply the field pair data if key is set; if not we just add the rule without it.